test/testtriag.jl

```diff
@@ -317,7 +317,7 @@ function test_triangular(elty1_types)
             @testset for elty2 in (Float32, Float64, BigFloat, ComplexF32, ComplexF64, Complex{BigFloat}, Int)
                 # Only test methods for the same element type and a single combination of mixed element types
                 # to avoid too much compilation
-                if !(elty1 == elty2 || elty1 ∈ (ComplexF32, Int) || elty1 ∈ (ComplexF32, Int))
+                if !(elty1 == elty2 || elty1 ∈ (ComplexF32, Int) || elty2 ∈ (ComplexF32, Int))
```

Comment:
What about

```suggestion
                if !(elty1 == elty2 || elty1 ∈ (ComplexF32, Int) && elty2 ∈ (ComplexF32, Int))
```

? Without deep thinking, just a quick suggestion.

Reply:
This restricts the `(elty1, elty2)` combinations for `Int` and `ComplexF32` types, while the floating-point types are only tested against themselves. Perhaps we may choose random combinations to test against.